From a551a3e506e18edd569745eb9190736b75899ee3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:57 -0400
Subject: [tools] perf/build: Fix timerfd feature check

Message-id: <1408462094-14194-309-git-send-email-jolsa@redhat.com>
Patchwork-id: 88305
O-Subject: [PATCH RHEL7.1 BZ1131394 308/325] tools/perf/build: Fix timerfd feature check
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit bb4c5500c9c5b809696eee212843e731721a6e15
Author: Ingo Molnar <mingo@kernel.org>
Date: Thu Nov 14 08:33:24 2013 +0100

description
===========
'feature_timerfd' is checked all the time and calculated explicitly,
in a serial fashion. Add it to CORE_FEATURE_TESTS which causes it to
be built in parallel, using the newfangled parallel build autodetection
code.

This shaves 137 msecs off the perf build time on my system, which
speeds up the common case cached build by 43%:

Before:

  comet:~/tip> perf stat --null --repeat 5 make -C tools/perf/
  [...]
         0,453771441 seconds time elapsed                                          ( +-  0,09% )

After:

  comet:~/tip> perf stat --null --repeat 5 make -C tools/perf/
  [...]
         0,316290185 seconds time elapsed                                          ( +-  0,24% )
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index f5905f2..861379e 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -142,7 +142,8 @@ CORE_FEATURE_TESTS =   \
  libunwind   \
  on-exit    \
  stackprotector   \
- stackprotector-all
+ stackprotector-all  \
+ timerfd
 
 #
 # So here we detect whether test-all was rebuilt, to be able
@@ -405,7 +406,6 @@ else
   endif
 endif
 
-$(call feature_check,timerfd)
 ifeq ($(feature-timerfd), 1)
   CFLAGS += -DHAVE_TIMERFD_SUPPORT
 else
-- 
1.7.1